core/transientstore/store.go

```diff
@@ -98,6 +114,173 @@ func (provider *storeProvider) Close() {
 	provider.dbProvider.Close()
 }
 
+// DeleteStore deletes the TransientStoreProvider for a given ledger.
+func (provider *storeProvider) DeleteStore(ledgerID string) error {
+	return provider.dbProvider.Drop(ledgerID)
+}
+
+func (provider *storeProvider) markStorageForDelete(ledgerID string) error {
+	marked, err := provider.getStorageMarkedForDeletion()
+	if err != nil {
+		return errors.WithMessage(err, "while listing delete marked storage")
+	}
+
+	// don't update if the storage is already marked for deletion.
+	for i := 0; i < len(marked); i++ {
+		if strings.EqualFold(ledgerID, marked[i]) {
+			logger.Infof("transient storage %s was already marked for delete", ledgerID)
```

Comment:
Here and other places
```suggestion
			logger.Infow("Transient storage was already marked for delete", "ledgerID", ledgerID)
```